Lib/argparse.py

```diff
@@ -2237,11 +2237,16 @@ def consume_positionals(start_index):
                     # twice (which may fail) if the argument was given, but
                     # only if it was defined already in the namespace
                     if (action.default is not None and
-                        isinstance(action.default, str) and
                         hasattr(namespace, action.dest) and
                         action.default is getattr(namespace, action.dest)):
-                        setattr(namespace, action.dest,
+                        if isinstance(action.default, str):
+                            setattr(namespace, action.dest,
                                 self._get_value(action, action.default))
+                        elif (action.nargs not in (OPTIONAL, None) and
+                              isinstance(action.default, (list, tuple)) and
```

Comment:
This seems like it's going to create issues. What is so special about `list` and `tuple`? What about set, deque, etc..? 

The design of argparse is very loose. I suspect "fixing" this without breaking backwards compatibility might be challenging.